kbi/0017/index.rst

```diff
@@ -0,0 +1,97 @@
+.. index::
+   single: <topic>; <subtopic>
+
+KBI0017: How to delete commits including their annexed data
+===========================================================
+
+:authors: Christian Mönch <christian.moench@web.de>
+:discussion: <link>
+:keywords: delete commits, delete annexed data
+:software-versions: datalad_0.8.13, git-annex_10.20230407
+
+This knowledge base item describes how to delete commits and
+the annexed data that was added to the annex when the commits
+were created.
+
+
+The initial state and the goal
+------------------------------
+
+There is a datalad dataset with an annex and a number of commits.
+Some of those commits added files to the annex, and some might
+just have added files to git. We assume that the last commit has
+added a number of files to the annex. The dataset has not yet been
+pushed to a sibling.
+
+The goal is to remove the last commit and all annexed files that
+were added when the commit was performed.
```

Comment:
I think the specific case from the user was actually not the last commit, but some commit before. Of course, the outlined process would not differ if it were the last commit or the 5th commit or whichever one, but I think the important distinction for the user was that they couldn't just use `git reset --hard` to a commit N, because that would remove all other commits that were made after N. That's why the `git rebase` was suggested in their case.

Whether or not, the example used in this KBI sticks with the last commit or uses an Nth commit, I think the point about `rebase` vs `reset` is worth making.